### DISCUSSION TITLE
How to use a component variable as model for monaco editor?

### DISCUSSION THREAD

**Discussion by soraphis** [django-commons/django-unicorn#610] [Q&A] [answered]
I'm currently struggling with it...

- when anything in the component changes the editor vanishes
- when I bind `monaco.editor.create` to the Unicorn "updated" event, I lose the cursor position in the editor

i don't need to pass the code all the time, but I have more things to the component.

**Comment by adamghill** [comment, marked as answer]
>when anything in the component changes the editor vanishes

This might be happening because the `morphdom` library is removing the editor. Have you tried ignoring that part of the component with https://www.django-unicorn.com/docs/templates/#ignore-elements?

>when I bind monaco.editor.create to the Unicorn "updated" event, I lose the cursor position in the editor

Hmm, I'm not sure what might be happening here. Can you make a repo with some code that replicates this so I could try it out?

**Reply by soraphis** [reply]
oh, thank you!
i guess with the ignore flag i can probably make it work good enough
